src/lib/caches/Items.js

```diff
@@ -94,22 +100,74 @@ module.exports = class ItemsCache {
       let exact = false;
       filtered = undefined;
       base
-        .filter((item) => item && item[by])
+        .filter((item) => item && this.#parseNestedKey(item, bys))
         .forEach((item) => {
-          if (item[by].toLowerCase() === term.toLowerCase()) {
-            filtered = item;
-            exact = true;
+          const values = this.#parseNestedKey(item, bys);
+          // parseNestedKey returns an array, so check all values in that array
+          for (const entry of values) {
+            //logger.error(entry.toLowerCase());
+            if (entry.toString().toLowerCase() === term.toString().toLowerCase()) {
+              filtered = item;
+              exact = true;
+            }
           }
```

Comment:
```suggestion
          values.forEach((entry) => {
            if (entry.toString().toLowerCase() === term.toString().toLowerCase()) {
              filtered = item;
              exact = true;
            }
          });
```

traditional for loops are general discouraged due to some extra runtimes they have to pull in